src/main/java/ca/on/oicr/pineryreports/reports/impl/GazpachoProjectStatusReport.java

```diff
@@ -0,0 +1,617 @@
+      if (lanesForCurrentRun == null) {
+        List<Integer> lanes = new ArrayList<>();
+        lanes.add(lane.getPosition());
+        lanesForRun.put(run.getId(), lanes);
+      } else {
+        lanesForRun.get(run.getId()).add(lane.getPosition());
+      }
+    }
+
+    public SampleDto getStock() {
+      return stock;
+    }
+
+    public Set<SampleDto> getAliquots() {
+      return aliquots;
+    }
+
+    public String getWaitingAliquots() {
+      Date receivedByLab;
+      try {
+        receivedByLab = getDateTimeFormat().parse(stock.getCreatedDate());
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for stock " + stock.getId());
+        return "";
+      }
+      if (aliquots.isEmpty() && receivedByLab.toInstant().isBefore(TWO_DAYS_AGO)) {
+        return "1";
+      } else {
+        return "";
+      }
+    }
+
+    public String getWaitingLibraries() {
+      Date receivedByLab;
+      SampleDto mostRecentAliquot = aliquots.stream().max(Comparator.comparing(SampleDto::getCreatedDate)).orElse(null);
+      if (mostRecentAliquot == null) return "";
+      try {
+        receivedByLab = getDateTimeFormat().parse(mostRecentAliquot.getCreatedDate());
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for aliquot " + mostRecentAliquot.getId());
+        return "";
+      }
+      if (getLibraries().isEmpty() && receivedByLab.toInstant().isBefore(TWO_DAYS_AGO)) {
+        return "1";
+      } else {
+        return "";
+      }
+    }
+
+    public String getExternalName() {
+      return externalName;
+    }
+
+    public String getGroupIds() {
+      String joined;
+      if (libraries.isEmpty()) {
+        joined = getAttribute(ATTR_GROUP_ID, stock);
+      } else {
+        joined = String.join(",", groupIds);
+      }
+      return (joined == null ? "" : joined);
+    }
+
+    public Set<SampleDto> getLibraries() {
+      return libraries;
+    }
+
+    public String getWaitingSequencing() {
+      if (getLibraries().isEmpty()) return "";
+      Date maxLibCreated = new Date();
+      try {
+        maxLibCreated = getDateTimeFormat().parse(getLibraries().stream()
+            .max(Comparator.comparing(SampleDto::getCreatedDate))
+            .map(SampleDto::getCreatedDate)
+            .orElse(new Date().toString()));
+      } catch (java.text.ParseException e) {
+        System.out.println("Error parsing created date for libraries for sample " + stock.getId());
+      }
+      if (runs.isEmpty() && maxLibCreated.toInstant().isBefore(TWO_DAYS_AGO)) {
+        return Integer.toString(libraries.size());
+      } else {
+        return "";
+      }
+    }
+
+    public String getNumLanesRunning() {
+      Set<RunDto> running = runs.stream().filter(run -> "Running".equals(run.getState())).collect(Collectors.toSet());
+      Integer lanesRunning = 0;
+      for (Integer runId : lanesForRun.keySet()) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Iterate over the "entrySet" instead of the "keySet". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2864)